extensions/wikia/PhalanxII/classes/PhalanxStatsPager.class.php

```diff
@@ -45,17 +45,28 @@ function getPagingQueries() {
 	}
 
 	function formatRow( $row ) {
-		$type = implode( ", ", Phalanx::getTypeNames( $row->ps_blocker_hit ?? $row->ps_blocker_type ) );
+		$blocker = $row->ps_blocker_hit ?: $row->ps_blocker_type;
+		$type = implode( ', ', Phalanx::getTypeNames( $blocker ) );
 		$username = $row->ps_blocked_user;
 		$timestamp = $this->getLanguage()->timeanddate( $row->ps_timestamp );
-		$oWiki = WikiFactory::getWikiByID( $row->ps_wiki_id );
-		$url = $row->ps_referrer ?? '';
-		$url = ( empty( $url ) && isset( $oWiki ) ) ? $oWiki->city_url : $url;
+
+		$url = $row->ps_referrer ?: '';
+		if ( empty( $url ) ) {
+			$wiki = WikiFactory::getWikiByID( $row->ps_wiki_id );
```

Comment:
We were doing this wiki factory look up every time previously even if we didn't need it because `$row->ps_referrer` was defined.  Only load wiki information from the DB if necessary.